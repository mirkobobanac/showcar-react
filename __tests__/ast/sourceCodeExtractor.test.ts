import * as ts from 'typescript'
import sourceCodeExtrator from '../../src/ast/sourceCodeExtractor'

const sourceFileMock = `
type MyType = {
  foo: boolean
}

const A = 'bar'

function C(s: string) {
 console.log(s)
}

/**
 * Some type declaration
 */
type Props = {
  /**
   * Some type comment
   */
  foo: number
}

/**
 * Some React component
 */
const SomeSFC = (p: Props) => (
  <div>
    {p.foo}
  </div>
)
`

describe('source code extraction', () => {
  test('extracts types aliases from a source code file', () => {
    const extracted = sourceCodeExtrator(sourceFileMock, [
      { kind: ts.SyntaxKind.TypeAliasDeclaration, value: 'Props' },
      { kind: ts.SyntaxKind.TypeAliasDeclaration, value: 'MyType' }
    ])

    expect(extracted).toMatchSnapshot()
  })

  test('extracts some React component from a source code file', () => {
    const extracted = sourceCodeExtrator(sourceFileMock, [{ kind: ts.SyntaxKind.VariableStatement, value: 'SomeSFC' }])

    expect(extracted).toMatchSnapshot()
  })
})
