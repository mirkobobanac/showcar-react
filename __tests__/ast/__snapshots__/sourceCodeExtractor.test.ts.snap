// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source code extraction extracts some React component from a source code file 1`] = `
"

/**
 * Some React component
 */
const SomeSFC = (p: Props) => (
  <div>
    {p.foo}
  </div>
)"
`;

exports[`source code extraction extracts types aliases from a source code file 1`] = `
"
type MyType = {
  foo: boolean
}

/**
 * Some type declaration
 */
type Props = {
  /**
   * Some type comment
   */
  foo: number
}"
`;
